File: 05 Database Bulk Update/TableColumnValueUpdate.py
import mysql.connector

# Your MySQL connection configuration
config = {
    'user': 'root',
    'password': '',
    'host': 'localhost',
    'database': 'property',
    'raise_on_warnings': True
}

# Create connection
conn = mysql.connector.connect(**config)

# Check connection
if conn.is_connected():
    print('Connected to MySQL database')

    # Select posts from wp_posts table where post_type is 'Property'
    cursor = conn.cursor()
    sql_select_posts = "SELECT ID FROM wp_posts WHERE post_type = 'Property'"
    cursor.execute(sql_select_posts)
    posts = cursor.fetchall()

    if posts:
        # Counter for serializing property_id
        counter = 1

        # Loop through each post
        for post in posts:
            post_id = post[0]

            # Serialize property_id
            new_property_id = 'TP_{:06d}'.format(counter)

            # Select from wp_postmeta table where post_id is $post_id and meta_key is '_property_property_id'
            sql_select_meta = "SELECT * FROM wp_postmeta WHERE post_id = %s AND meta_key = '_property_property_id'"
            cursor.execute(sql_select_meta, (post_id,))
            meta_result = cursor.fetchall()

            if meta_result:
                # Update the unique meta_value into wp_postmeta table
                sql_update_meta = "UPDATE wp_postmeta SET meta_value=%s WHERE post_id = %s AND meta_key = '_property_property_id'"
                cursor.execute(sql_update_meta, (new_property_id, post_id))
                conn.commit()
                print("Record updated successfully")
            else:
                # Insert the unique meta_value into wp_postmeta table
                sql_insert_meta = "INSERT INTO wp_postmeta (post_id, meta_key, meta_value) VALUES (%s, '_property_property_id', %s)"
                cursor.execute(sql_insert_meta, (post_id, new_property_id))
                conn.commit()
                print("New record inserted successfully")

            # Increment the counter for the next iteration
            counter += 1

    else:
        print("0 results found in wp_posts table")

    # Close cursor and connection
    cursor.close()
    conn.close()
    print('MySQL connection closed')
else:
    print('Connection to MySQL database failed')

File: Project/Database Bulk Update/TableColumnValueUpdate.php
<?php

	// Your MySQL connection configuration
	$servername = "localhost";
	$username   = "root";
	$password   = "";
	$dbname     = "property";

	// Create connection
	$conn = new mysqli($servername, $username, $password, $dbname);

	// Check connection
	if ($conn->connect_error) {
		die("Connection failed: " . $conn->connect_error);
	}

	// Select posts from wp_posts table where post_type is 'Property'
	$sql_select_posts = "SELECT ID FROM wp_posts WHERE post_type = 'Property'";
	$result_posts     = $conn->query($sql_select_posts);

	if ($result_posts->num_rows > 0) {
		// Counter for serializing property_id
		$counter = 1;

		// Loop through each post
		while($row_posts = $result_posts->fetch_assoc()) {
			$post_id = $row_posts['ID'];
			
			// Serialize property_id
			$new_property_id = 'TP_' . sprintf('%06d', $counter);
			
			// Select * from wp_postmeta table where post_id is $post_id and meta_key is '_property_property_id'
			$sql_select_metas = "SELECT * FROM wp_postmeta WHERE post_id = $post_id AND meta_key = '_property_property_id'";
			$result_metas = $conn->query($sql_select_metas);
			if ($result_metas->num_rows > 0) {
				// Updated the unique meta_value into wp_postmeta table
				$sql_update_postmeta = "UPDATE wp_postmeta SET meta_value='$new_property_id' WHERE post_id = $post_id AND meta_key = '_property_property_id'";
				if ($conn->query($sql_update_postmeta) === TRUE) {
					echo "Record updated successfully";
				} else {
					echo "Error updating record: " . $conn->error;
				}
			} else {
				// Insert the unique meta_value into wp_postmeta table
				$sql_insert_postmeta = "INSERT INTO wp_postmeta (post_id, meta_key, meta_value) VALUES ($post_id, '_property_property_id', '$new_property_id')";
				if ($conn->query($sql_insert_postmeta) === TRUE) {
					echo "New record inserted successfully";
				} else {
					echo "Error inserting record: " . $conn->error;
				}
			}           
			
			// Increment the counter for the next iteration
			$counter++;
		}
	} else {
		echo "0 results found in wp_posts table";
	}

	// Close MySQL connection
	$conn->close();

?>
